dom.go

```diff
@@ -124,8 +141,40 @@ func (h *HTML) restoreText(prev *HTML) {
 	}
 }
 
-func (h *HTML) restoreHTML(prev *HTML) {
-	h.node = prev.node
+func (h *HTML) mutate(prev *HTML) {
```

Comment:
I think that `reconcile` would be a better name for this, since it's reconciling state from a new `*HTML` against a prev / old `*HTML`. In fact, this was the original name before we renamed it to 'restore' in an attempt to shorten the (external) name. Since it's internal, though, I feel `reconcile` would do better than `mutate`. What do you think?